src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,66 @@
 package com.theironyard.novauc;
 
+import spark.ModelAndView;
+import spark.Session;
+import spark.Spark;
+import spark.template.mustache.MustacheTemplateEngine;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
 public class Main {
 
+    static HashMap<String, User> users = new HashMap<>();
+    static ArrayList<Message> messages = new ArrayList<>();
+
     public static void main(String[] args) {
-	// write your code here
+
+        Spark.init();
+        Spark.get(
+                "/",
+                ((request, response) -> {
+                    HashMap m = new HashMap();
+                    ArrayList<Message> threads = new ArrayList<>();
+                    for (Message message : messages) {
+                        if (message.replyId == -1) {
+                            threads.add(message);
+                        }
+                    }
+
+
+                    Session session = request.session();
+                    String userName = session.attribute("userName");
+
+                    m.put("messages", threads);
+                    m.put("userName", userName);
+                    return new ModelAndView(m, "home.html");
+                }),
+                new MustacheTemplateEngine()
+        );
+
+
+        Spark.post(
+                "/login",
+                ((request, response) -> {
+                    String userName = request.queryParams("loginName");
+                    if (userName == null) {
+                        throw new Exception("Login name not found.");
+                    }
+
+                    User user = users.get(userName);
+                    if (user == null) {
+                        user = new User(userName);
+                        users.put(userName, user);
+                    }
+
+                    Session session = request.session();
+                    session.attribute("userName", userName);
+
+                    response.redirect("/");
+                    return "";
+                })
+        );
```

Comment:
Login looks good.  Html looks good as well.  But the rest of the code isn't really going to be needed for this assignment.